odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -464,6 +465,70 @@ def send_email(self, message, mail_server_id=None, smtp_server=None, smtp_port=N
             raise MailDeliveryException(_("Mail Delivery Failed"), msg)
         return message_id
 
+    @api.multi
+    def _get_canonical_from(self, original_from):
+        """Returns a FROM header to respect the SMTP server settings.
+
+        Args:
+            original_from (str): The original FROM header for the message.
+
+        Returns:
+            str: A string that should be used in a message header.
+        """
+        # Use the canonical name, if existing.
+        address = parseaddr(original_from)
+        # Otherwise we use the name part of the email address.
+        if not address[0]:
+            address[0] = address[1].split('@', 1)[0]
+        return '%s via %s <%s>' % (
+            address[0],
+            self.env.user.company_id.name,
+            self._get_default_bounce_address(),
```

Comment:
bounce address or catchall?

Reply:
I think these two concepts are the same? The bounce address is a combination of the catchall domain, plus the bounce alias. The description in the doc block seems on par with what we're going for:

> The default bounce address is used to set the envelop address if no
> envelop address is provided in the message.  It is formed by properly
> joining the parameters "mail.catchall.alias" and
> "mail.catchall.domain".

Reply:
Ah, OK, thanks for the clarification. I remember that there were a specific system parameter for putting another bounce alias, but now it seems it has gone out.